packages/runtime/container-runtime/src/blobManager/blobManager.ts

```diff
@@ -65,17 +68,29 @@ import {
  */
 export class BlobHandle
 	extends FluidHandleBase<ArrayBufferLike>
-	implements IFluidHandleInternalPayloadPending<ArrayBufferLike>
+	implements
+		IFluidHandlePayloadPending<ArrayBufferLike>,
+		IFluidHandleInternalPayloadPending<ArrayBufferLike>
 {
 	private attached: boolean = false;
 
 	public get isAttached(): boolean {
 		return this.routeContext.isAttached && this.attached;
 	}
 
+	private readonly _events = new TypedEventEmitter<IFluidHandlePayloadPendingEvents>();
```

Comment:
i have some concerns about memory usage. a simple callback would be much cheaper in terms of memory, especially as we don't think will be used very often, as after a handle is in a terminal state there will never be a need to listen to events again

Reply:
another middle ground option would be to lazily initialize the TypedEventEmitter, so it only exists when it is accessed. emission would need to change to not create it if it doesn't already exist

Reply:
Will look at options here.  I was also thinking I should probably look at switching to the new Tree emitter too.

Reply:
Changed in latest.  Realistically it probably doesn't matter much here since the handle implies we're holding the blob itself in memory which probably dwarfs the emitter's empty Map, but might as well.